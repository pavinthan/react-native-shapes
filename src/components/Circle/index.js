import Circle from './Circle';
import defaultProps from './defaultProps';
import propTypes from './propTypes';

Circle.propTypes = propTypes;
Circle.defaultProps = defaultProps;

export default Circle;
