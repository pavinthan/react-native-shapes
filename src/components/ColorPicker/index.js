import ColorPicker from './ColorPicker';
import defaultProps from './defaultProps';
import propTypes from './propTypes';

ColorPicker.propTypes = propTypes;
ColorPicker.defaultProps = defaultProps;

export default ColorPicker;
