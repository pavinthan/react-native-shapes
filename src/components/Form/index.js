import Form from './Form';
import defaultProps from './defaultProps';
import propTypes from './propTypes';

Form.propTypes = propTypes;
Form.defaultProps = defaultProps;

export default Form;
