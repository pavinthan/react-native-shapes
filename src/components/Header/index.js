import Header from './Header';
import defaultProps from './defaultProps';
import propTypes from './propTypes';

Header.propTypes = propTypes;
Header.defaultProps = defaultProps;

export default Header;
