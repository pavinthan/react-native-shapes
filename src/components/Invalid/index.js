import Invalid from './Invalid';
import defaultProps from './defaultProps';
import propTypes from './propTypes';

Invalid.propTypes = propTypes;
Invalid.defaultProps = defaultProps;

export default Invalid;
