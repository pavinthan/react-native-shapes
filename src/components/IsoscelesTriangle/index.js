import IsoscelesTriangle from './IsoscelesTriangle';
import defaultProps from './defaultProps';
import propTypes from './propTypes';

IsoscelesTriangle.propTypes = propTypes;
IsoscelesTriangle.defaultProps = defaultProps;

export default IsoscelesTriangle;
