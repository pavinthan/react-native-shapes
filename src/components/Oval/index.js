import Oval from './Oval';
import defaultProps from './defaultProps';
import propTypes from './propTypes';

Oval.propTypes = propTypes;
Oval.defaultProps = defaultProps;

export default Oval;
