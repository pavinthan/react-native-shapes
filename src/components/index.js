export { default as Circle } from './Circle';
export { default as ColorPicker } from './ColorPicker';
export { default as Form } from './Form';
export { default as Header } from './Header';
export { default as Invalid } from './Invalid';
export { default as IsoscelesTriangle } from './IsoscelesTriangle';
export { default as Oval } from './Oval';
export { default as Parallelogram } from './Parallelogram';
export { default as Polygon } from './Polygon';
export { default as Rectangle } from './Rectangle';
export { default as Whiteboard } from './Whiteboard';
