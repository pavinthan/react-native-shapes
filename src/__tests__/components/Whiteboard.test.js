import React from 'react';
import { Whiteboard } from '../../components';
import renderer from 'react-test-renderer';
import {
  CIRCLE,
  HEPTAGON,
  HEXAGON,
  EQUILATERAL_TRIANGLE,
  ISOSCELES_TRIANGLE,
  SCALENE_TRIANGLE,
  OCTAGON,
  OVAL,
  PARALLELOGRAM,
  PENTAGON,
  RECTANGLE,
  SQUARE,
} from '../../constants';

describe('circle component', () => {
  it('renders without crashing', () => {
    const rendered = renderer
      .create(
        <Whiteboard shape={CIRCLE} measurements={['radius']} values={[100]} />
      )
      .toJSON();
    expect(rendered).toBeTruthy();
  });

  it('renders component', () => {
    const tree = renderer
      .create(
        <Whiteboard shape={CIRCLE} measurements={['radius']} values={[100]} />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('oval component', () => {
  it('renders without crashing', () => {
    const rendered = renderer
      .create(
        <Whiteboard
          shape={OVAL}
          measurements={['width', 'height']}
          values={[200, 100]}
        />
      )
      .toJSON();
    expect(rendered).toBeTruthy();
  });

  it('renders component', () => {
    const tree = renderer
      .create(
        <Whiteboard
          shape={OVAL}
          measurements={['width', 'height']}
          values={[200, 100]}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('equilateral triangle component', () => {
  it('renders without crashing', () => {
    const rendered = renderer
      .create(
        <Whiteboard
          shape={EQUILATERAL_TRIANGLE}
          measurements={['sideLength']}
          values={[100]}
        />
      )
      .toJSON();
    expect(rendered).toBeTruthy();
  });

  it('renders component', () => {
    const tree = renderer
      .create(
        <Whiteboard
          shape={EQUILATERAL_TRIANGLE}
          measurements={['sideLength']}
          values={[100]}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('isosceles triangle component', () => {
  it('renders without crashing', () => {
    const rendered = renderer
      .create(
        <Whiteboard
          shape={ISOSCELES_TRIANGLE}
          measurements={['width', 'height']}
          values={[100, 200]}
        />
      )
      .toJSON();
    expect(rendered).toBeTruthy();
  });

  it('renders component', () => {
    const tree = renderer
      .create(
        <Whiteboard
          shape={ISOSCELES_TRIANGLE}
          measurements={['width', 'height']}
          values={[100, 200]}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('scalene triangle component', () => {
  it('renders without crashing', () => {
    const rendered = renderer
      .create(
        <Whiteboard
          shape={SCALENE_TRIANGLE}
          measurements={['width', 'height']}
          values={[100, 200]}
        />
      )
      .toJSON();
    expect(rendered).toBeTruthy();
  });

  it('renders component', () => {
    const tree = renderer
      .create(
        <Whiteboard
          shape={SCALENE_TRIANGLE}
          measurements={['width', 'height']}
          values={[100, 200]}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('rectangle component', () => {
  it('renders without crashing', () => {
    const rendered = renderer
      .create(
        <Whiteboard
          shape={RECTANGLE}
          measurements={['width', 'height']}
          values={[100, 200]}
        />
      )
      .toJSON();
    expect(rendered).toBeTruthy();
  });

  it('renders component', () => {
    const tree = renderer
      .create(
        <Whiteboard
          shape={RECTANGLE}
          measurements={['width', 'height']}
          values={[100, 200]}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('parallelogram component', () => {
  it('renders without crashing', () => {
    const rendered = renderer
      .create(
        <Whiteboard
          shape={PARALLELOGRAM}
          measurements={['width', 'height']}
          values={[100, 200]}
        />
      )
      .toJSON();
    expect(rendered).toBeTruthy();
  });

  it('renders component', () => {
    const tree = renderer
      .create(
        <Whiteboard
          shape={PARALLELOGRAM}
          measurements={['width', 'height']}
          values={[100, 200]}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('square component', () => {
  it('renders without crashing', () => {
    const rendered = renderer
      .create(
        <Whiteboard
          shape={SQUARE}
          measurements={['side length']}
          values={[100]}
        />
      )
      .toJSON();
    expect(rendered).toBeTruthy();
  });

  it('renders component', () => {
    const tree = renderer
      .create(
        <Whiteboard
          shape={SQUARE}
          measurements={['side length']}
          values={[100]}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('pentagon component', () => {
  it('renders without crashing', () => {
    const rendered = renderer
      .create(
        <Whiteboard
          shape={PENTAGON}
          measurements={['side length']}
          values={[100]}
        />
      )
      .toJSON();
    expect(rendered).toBeTruthy();
  });

  it('renders component', () => {
    const tree = renderer
      .create(
        <Whiteboard
          shape={PENTAGON}
          measurements={['side length']}
          values={[100]}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('heptagon component', () => {
  it('renders without crashing', () => {
    const rendered = renderer
      .create(
        <Whiteboard
          shape={HEPTAGON}
          measurements={['sideLength']}
          values={[100]}
        />
      )
      .toJSON();
    expect(rendered).toBeTruthy();
  });

  it('renders component', () => {
    const tree = renderer
      .create(
        <Whiteboard
          shape={HEPTAGON}
          measurements={['sideLength']}
          values={[100]}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('hexagon component', () => {
  it('renders without crashing', () => {
    const rendered = renderer
      .create(
        <Whiteboard
          shape={HEXAGON}
          measurements={['sideLength']}
          values={[100]}
        />
      )
      .toJSON();
    expect(rendered).toBeTruthy();
  });

  it('renders component', () => {
    const tree = renderer
      .create(
        <Whiteboard
          shape={HEXAGON}
          measurements={['sideLength']}
          values={[100]}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('octagon component', () => {
  it('renders without crashing', () => {
    const rendered = renderer
      .create(
        <Whiteboard
          shape={OCTAGON}
          measurements={['sideLength']}
          values={[100]}
        />
      )
      .toJSON();
    expect(rendered).toBeTruthy();
  });

  it('renders component', () => {
    const tree = renderer
      .create(
        <Whiteboard
          shape={OCTAGON}
          measurements={['sideLength']}
          values={[100]}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('invalid', () => {
  it('renders invalid without crashing', () => {
    const rendered = renderer
      .create(<Whiteboard shape="example" measurements={[]} values={[]} />)
      .toJSON();
    expect(rendered).toBeTruthy();
  });

  it('renders whiteboard invalid component', () => {
    const tree = renderer
      .create(<Whiteboard shape="example" measurements={[]} values={[]} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
