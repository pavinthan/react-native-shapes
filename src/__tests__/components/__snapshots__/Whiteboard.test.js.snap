// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`circle component renders component 1`] = `
<View
  style={
    Array [
      Object {},
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
    ]
  }
  virtual={undefined}
>
  <RNSVGSvgView
    bbHeight="200"
    bbWidth="200"
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
        false,
        Object {
          "flex": 0,
          "height": 200,
          "width": 200,
        },
      ]
    }
  >
    <RNSVGCircle
      cx="100"
      cy="100"
      fill={
        Array [
          0,
          0,
          0,
          0,
          1,
        ]
      }
      fillOpacity={1}
      fillRule={1}
      matrix={
        Array [
          1,
          0,
          0,
          1,
          0,
          0,
        ]
      }
      onLayout={undefined}
      opacity={1}
      originX={0}
      originY={0}
      propList={
        Array [
          "fill",
        ]
      }
      r="100"
      rotation={0}
      scaleX={1}
      scaleY={1}
      skewX={0}
      skewY={0}
      stroke={null}
      strokeDasharray={null}
      strokeDashoffset={null}
      strokeLinecap={0}
      strokeLinejoin={0}
      strokeMiterlimit={4}
      strokeOpacity={1}
      strokeWidth="1"
      x={0}
      y={0}
    />
  </RNSVGSvgView>
</View>
`;

exports[`circle component renders component 2`] = `
<View
  style={
    Array [
      Object {},
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
    ]
  }
  virtual={undefined}
>
  <RNSVGSvgView
    bbHeight="200"
    bbWidth="400"
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
        false,
        Object {
          "flex": 0,
          "height": 200,
          "width": 400,
        },
      ]
    }
  >
    <RNSVGEllipse
      cx="200"
      cy="100"
      fill={
        Array [
          0,
          0,
          0,
          0,
          1,
        ]
      }
      fillOpacity={1}
      fillRule={1}
      matrix={
        Array [
          1,
          0,
          0,
          1,
          0,
          0,
        ]
      }
      onLayout={undefined}
      opacity={1}
      originX={0}
      originY={0}
      propList={
        Array [
          "fill",
        ]
      }
      rotation={0}
      rx="100"
      ry="50"
      scaleX={1}
      scaleY={1}
      skewX={0}
      skewY={0}
      stroke={null}
      strokeDasharray={null}
      strokeDashoffset={null}
      strokeLinecap={0}
      strokeLinejoin={0}
      strokeMiterlimit={4}
      strokeOpacity={1}
      strokeWidth="1"
      x={0}
      y={0}
    />
  </RNSVGSvgView>
</View>
`;

exports[`heptagon component renders component 1`] = `
<View
  style={
    Array [
      Object {},
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
    ]
  }
  virtual={undefined}
>
  <ARTSurfaceView
    style={
      Array [
        undefined,
        Object {
          "height": 300,
          "width": 300,
        },
      ]
    }
  >
    <ARTGroup
      opacity={1}
      transform={
        Array [
          1,
          0,
          0,
          1,
          100,
          33.333333333333336,
        ]
      }
    >
      <ARTShape
        d={
          Array [
            0,
            100,
            0,
            2,
            162.34898018587336,
            78.18314824680299,
            2,
            140.0968867902419,
            175.67593946498533,
            2,
            50,
            219.06431337674115,
            2,
            -40.09688679024191,
            175.67593946498536,
            2,
            -62.34898018587337,
            78.183148246803,
            2,
            -3.552713678800501e-14,
            1.4210854715202004e-14,
            2,
            99.99999999999997,
            -1.028208126774506e-14,
            1,
          ]
        }
        fill={
          Array [
            0,
            0,
            0,
            0,
            1,
          ]
        }
        opacity={1}
        stroke={null}
        strokeCap={1}
        strokeDash={null}
        strokeJoin={1}
        strokeWidth={1}
        transform={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
      />
    </ARTGroup>
  </ARTSurfaceView>
</View>
`;

exports[`hexagon component renders component 1`] = `
<View
  style={
    Array [
      Object {},
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
    ]
  }
  virtual={undefined}
>
  <ARTSurfaceView
    style={
      Array [
        undefined,
        Object {
          "height": 300,
          "width": 300,
        },
      ]
    }
  >
    <ARTGroup
      opacity={1}
      transform={
        Array [
          1,
          0,
          0,
          1,
          100,
          33.333333333333336,
        ]
      }
    >
      <ARTShape
        d={
          Array [
            0,
            100,
            0,
            2,
            150,
            86.60254037844386,
            2,
            100.00000000000003,
            173.20508075688775,
            2,
            2.842170943040401e-14,
            173.20508075688775,
            2,
            -50.000000000000014,
            86.6025403784439,
            2,
            0,
            4.263256414560601e-14,
            2,
            100,
            1.8139628162658947e-14,
            1,
          ]
        }
        fill={
          Array [
            0,
            0,
            0,
            0,
            1,
          ]
        }
        opacity={1}
        stroke={null}
        strokeCap={1}
        strokeDash={null}
        strokeJoin={1}
        strokeWidth={1}
        transform={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
      />
    </ARTGroup>
  </ARTSurfaceView>
</View>
`;

exports[`invalid component renders whiteboard invalid component 1`] = `
<View
  style={
    Array [
      Object {},
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
    ]
  }
  virtual={undefined}
>
  <View
    style={
      Array [
        Object {},
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        },
      ]
    }
    virtual={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Array [
            Object {
              "color": "#000",
              "fontSize": 30,
            },
            Object {
              "color": "#000",
              "fontSize": 30,
            },
            Object {
              "fontSize": 33,
              "marginBottom": 10,
            },
            Object {
              "color": "#000",
            },
          ],
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
      type="FontAwesome"
      virtual={undefined}
    >
      ÔÅ™
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 17,
            "fontWeight": "700",
            "marginLeft": undefined,
            "paddingLeft": 4,
            "paddingTop": 1,
            "textAlign": "center",
          },
          undefined,
          Object {
            "color": "#000",
          },
        ]
      }
      virtual={undefined}
    >
      Invalid Shape
    </Text>
  </View>
</View>
`;

exports[`isosceles triangle component renders component 1`] = `
<View
  style={
    Array [
      Object {},
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
    ]
  }
  virtual={undefined}
>
  <RNSVGSvgView
    bbHeight="200"
    bbWidth="100"
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
        false,
        Object {
          "flex": 0,
          "height": 200,
          "width": 100,
        },
      ]
    }
  >
    <RNSVGPath
      d="M0 0 0 100 100 0z"
      fill={
        Array [
          0,
          0,
          0,
          0,
          1,
        ]
      }
      fillOpacity={1}
      fillRule={1}
      matrix={
        Array [
          1,
          0,
          0,
          1,
          0,
          0,
        ]
      }
      onLayout={undefined}
      opacity={1}
      originX={0}
      originY={0}
      propList={
        Array [
          "fill",
        ]
      }
      rotation={0}
      scaleX={1}
      scaleY={1}
      skewX={0}
      skewY={0}
      stroke={null}
      strokeDasharray={null}
      strokeDashoffset={null}
      strokeLinecap={0}
      strokeLinejoin={0}
      strokeMiterlimit={4}
      strokeOpacity={1}
      strokeWidth="1"
      x={0}
      y={0}
    />
  </RNSVGSvgView>
</View>
`;

exports[`octagon component renders component 1`] = `
<View
  style={
    Array [
      Object {},
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
    ]
  }
  virtual={undefined}
>
  <ARTSurfaceView
    style={
      Array [
        undefined,
        Object {
          "height": 300,
          "width": 300,
        },
      ]
    }
  >
    <ARTGroup
      opacity={1}
      transform={
        Array [
          1,
          0,
          0,
          1,
          100,
          33.333333333333336,
        ]
      }
    >
      <ARTShape
        d={
          Array [
            0,
            100,
            0,
            2,
            170.71067811865476,
            70.71067811865474,
            2,
            170.71067811865476,
            170.71067811865476,
            2,
            100.00000000000001,
            241.4213562373095,
            2,
            1.4210854715202004e-14,
            241.4213562373095,
            2,
            -70.71067811865476,
            170.71067811865476,
            2,
            -70.71067811865477,
            70.71067811865476,
            2,
            -2.842170943040401e-14,
            -1.4210854715202004e-14,
            2,
            99.99999999999997,
            -3.870379069814907e-14,
            1,
          ]
        }
        fill={
          Array [
            0,
            0,
            0,
            0,
            1,
          ]
        }
        opacity={1}
        stroke={null}
        strokeCap={1}
        strokeDash={null}
        strokeJoin={1}
        strokeWidth={1}
        transform={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
      />
    </ARTGroup>
  </ARTSurfaceView>
</View>
`;

exports[`parallelogram component renders component 1`] = `
<View
  style={
    Array [
      Object {},
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
    ]
  }
  virtual={undefined}
>
  <ARTSurfaceView
    style={
      Array [
        undefined,
        Object {
          "height": 200,
          "width": 150,
        },
      ]
    }
  >
    <ARTGroup
      opacity={1}
      transform={
        Array [
          1,
          0,
          0,
          1,
          0,
          0,
        ]
      }
    >
      <ARTShape
        d={
          Array [
            0,
            50,
            0,
            2,
            150,
            0,
            2,
            100,
            200,
            2,
            0,
            200,
            2,
            50,
            0,
            1,
          ]
        }
        fill={
          Array [
            0,
            0,
            0,
            0,
            1,
          ]
        }
        opacity={1}
        stroke={null}
        strokeCap={1}
        strokeDash={null}
        strokeJoin={1}
        strokeWidth={1}
        transform={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
      />
    </ARTGroup>
  </ARTSurfaceView>
</View>
`;

exports[`pentagon component renders component 1`] = `
<View
  style={
    Array [
      Object {},
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
    ]
  }
  virtual={undefined}
>
  <ARTSurfaceView
    style={
      Array [
        undefined,
        Object {
          "height": 300,
          "width": 300,
        },
      ]
    }
  >
    <ARTGroup
      opacity={1}
      transform={
        Array [
          1,
          0,
          0,
          1,
          100,
          33.333333333333336,
        ]
      }
    >
      <ARTShape
        d={
          Array [
            0,
            100,
            0,
            2,
            130.90169943749476,
            95.10565162951535,
            2,
            50.00000000000003,
            153.88417685876266,
            2,
            -30.901699437494713,
            95.10565162951536,
            2,
            1.0658141036401503e-14,
            0,
            2,
            100.00000000000001,
            -2.4492935982947064e-14,
            1,
          ]
        }
        fill={
          Array [
            0,
            0,
            0,
            0,
            1,
          ]
        }
        opacity={1}
        stroke={null}
        strokeCap={1}
        strokeDash={null}
        strokeJoin={1}
        strokeWidth={1}
        transform={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
      />
    </ARTGroup>
  </ARTSurfaceView>
</View>
`;

exports[`rectangle component renders component 1`] = `
<View
  style={
    Array [
      Object {},
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
    ]
  }
  virtual={undefined}
>
  <RNSVGSvgView
    bbHeight="200"
    bbWidth="100"
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
        false,
        Object {
          "flex": 0,
          "height": 200,
          "width": 100,
        },
      ]
    }
  >
    <RNSVGRect
      fill={
        Array [
          0,
          0,
          0,
          0,
          1,
        ]
      }
      fillOpacity={1}
      fillRule={1}
      height="200"
      matrix={
        Array [
          1,
          0,
          0,
          1,
          0,
          0,
        ]
      }
      onLayout={undefined}
      opacity={1}
      originX={0}
      originY={0}
      propList={
        Array [
          "fill",
        ]
      }
      rotation={0}
      rx="0"
      ry="0"
      scaleX={1}
      scaleY={1}
      skewX={0}
      skewY={0}
      stroke={null}
      strokeDasharray={null}
      strokeDashoffset={null}
      strokeLinecap={0}
      strokeLinejoin={0}
      strokeMiterlimit={4}
      strokeOpacity={1}
      strokeWidth="1"
      width="100"
      x="0"
      y="0"
    />
  </RNSVGSvgView>
</View>
`;

exports[`square component renders component 1`] = `
<View
  style={
    Array [
      Object {},
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
    ]
  }
  virtual={undefined}
>
  <ARTSurfaceView
    style={
      Array [
        undefined,
        Object {
          "height": 300,
          "width": 300,
        },
      ]
    }
  >
    <ARTGroup
      opacity={1}
      transform={
        Array [
          1,
          0,
          0,
          1,
          100,
          33.333333333333336,
        ]
      }
    >
      <ARTShape
        d={
          Array [
            0,
            100,
            0,
            2,
            100,
            100,
            2,
            0,
            100.00000000000001,
            2,
            -1.8369701987210297e-14,
            1.4210854715202004e-14,
            2,
            99.99999999999999,
            -1.028208126774506e-14,
            1,
          ]
        }
        fill={
          Array [
            0,
            0,
            0,
            0,
            1,
          ]
        }
        opacity={1}
        stroke={null}
        strokeCap={1}
        strokeDash={null}
        strokeJoin={1}
        strokeWidth={1}
        transform={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
      />
    </ARTGroup>
  </ARTSurfaceView>
</View>
`;
